frontend/src/metabase/query_builder/components/filters/modals/InlineValuePicker/InlineValuePicker.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useCallback, useMemo } from "react";
+import { t } from "ttag";
+import Filter from "metabase-lib/lib/queries/structured/Filter";
+import Field from "metabase-lib/lib/metadata/Field";
+
+import { ALLOWED_OPERATORS } from "./constants";
+import { OperatorSelector, ArgumentSelector } from "./InlineValuePicker.styled";
+
+interface InlineValuePickerProps {
+  filter: Filter;
+  field: Field;
+  handleChange: (newFilter: Filter) => void;
+}
+
+export function InlineValuePicker({
+  filter,
+  field,
+  handleChange,
+}: InlineValuePickerProps) {
+  const changeOperator = useCallback(
+    (newOperator: any) => {
+      handleChange(filter.setOperator(newOperator));
+    },
+    [filter, handleChange],
+  );
+
+  const changeArguments = useCallback(
+    (newArguments: number[]) => {
+      handleChange(filter.setArguments(newArguments));
+    },
+    [filter, handleChange],
+  );
+
+  const filterOperators = useMemo(() => {
+    return field.filterOperators(filter.operatorName());
+  }, [field, filter]);
```

Comment:
So I just figure out how `memoizeClass` works today. Since `filterOperators` is already memoized here.

https://github.com/metabase/metabase/blob/36ae31ea30adf332a05f4f7b4a93a63b8c57827d/frontend/src/metabase-lib/lib/metadata/Field.ts#L457

Should we memoize this again? It seems redundant, but I also know it's not quite obvious `filterOperators` is already memoize at glance.

Reply:
I think it probably doesn't need to be memoized for a different reason - one of its dependencies will likely be changing on every rerender anyway.